packages/appeals-service-api/src/repositories/document-metadata-repository.js

```diff
@@ -0,0 +1,37 @@
+const { MongoRepository } = require('./mongo-repository');
+
+class DocumentMetadataRepository extends MongoRepository {
+	constructor() {
+		super('documentMetadata');
+	}
+	/**
+	 * @typedef {'Supporting Documents' | 'Appeal Statement' | 'Planning application form' | 'Decision notice'} DocumentType
+	 */
+
+	/**
+	 *
+	 * @param {string} caseRef
+	 * @param {DocumentType} documentType
+	 * @return {Promise<any>}
+	 */
+	async getDocumentMetadata(caseRef, documentType) {
+		try {
+			return await this.findOneByQuery(
+				{
+					caseRef: caseRef,
+					documentType: documentType,
+					virusCheckStatus: 'scanned',
+					redactedStatus: 'redacted',
+					publishedStatus: 'published'
+				},
+				{
+					version: -1
```

Comment:
Nice elegant query, only think I would say is it might make unit testing a bit easier if the different statuses were constants, but we're not planning on using them anywhere else so I'm not too fussed if we keep them as inline strings.

Reply:
Unit testing? Where? 😷

Reply:
I guess when we write the client for these functions there might be some mock document metadata with statuses that will be the same as these?

Reply:
It's more than likely. Is there anything to do for this ticket?

Reply:
No all good 👍